src/Translator/assets/translator-types-tests.ts

```diff
@@ -0,0 +1,80 @@
+
+trans(MESSAGE_MULTI_DOMAINS);
+trans(MESSAGE_MULTI_DOMAINS, {});
+trans(MESSAGE_MULTI_DOMAINS, {}, 'foobar');
+trans(MESSAGE_MULTI_DOMAINS, {}, 'foobar', 'fr');
+trans(MESSAGE_MULTI_DOMAINS, {}, 'messages');
+trans(MESSAGE_MULTI_DOMAINS, {}, 'messages', 'fr');
+
+const MESSAGE_MULTI_DOMAINS_WITH_PARAMETERS: Message<{ foobar: { parameters: { parameter1: string } }, messages: { parameters: { parameter2: string } } }, 'fr'> = {
+    id: 'message.multi.domains.with.parameters',
+    translations: {
+        foobar: {
+            fr: 'Un simple message {parameter1}',
+        },
+        messages: {
+            fr: 'Un simple message {parameter2}',
+        }
+    }
+};
+
+trans(MESSAGE_MULTI_DOMAINS_WITH_PARAMETERS, {parameter1: 'foo', parameter2: 'bar'});
+trans(MESSAGE_MULTI_DOMAINS_WITH_PARAMETERS, {parameter1: 'foo', parameter2: 'bar'}, 'foobar');
+trans(MESSAGE_MULTI_DOMAINS_WITH_PARAMETERS, {parameter1: 'foo', parameter2: 'bar'}, 'foobar', 'fr');
+trans(MESSAGE_MULTI_DOMAINS_WITH_PARAMETERS, {parameter1: 'foo', parameter2: 'bar'}, 'messages');
+trans(MESSAGE_MULTI_DOMAINS_WITH_PARAMETERS, {parameter1: 'foo', parameter2: 'bar'}, 'messages', 'fr');
+
+trans(MESSAGE_MULTI_DOMAINS_WITH_PARAMETERS, {parameter1: 'foo'}, 'qsd');
+
+const MESSAGE_MULTI_DOMAINS_MULTI_LOCALES: Message<{ foobar: { parameters: NoParametersType }, messages: { parameters: NoParametersType } }, 'fr' | 'en'> = {
+    id: 'message.multi.domains.multi.locales',
+    translations: {
+        foobar: {
+            fr: 'Un simple message',
+            en: 'A simple message',
+        },
+        messages: {
+            fr: 'Un simple message',
+            en: 'A simple message',
+        }
+    }
+};
+
+trans(MESSAGE_MULTI_DOMAINS_MULTI_LOCALES);
+trans(MESSAGE_MULTI_DOMAINS_MULTI_LOCALES, {},);
+trans(MESSAGE_MULTI_DOMAINS_MULTI_LOCALES, {}, 'messages');
+trans(MESSAGE_MULTI_DOMAINS_MULTI_LOCALES, {}, 'messages', 'fr');
+trans(MESSAGE_MULTI_DOMAINS_MULTI_LOCALES, {}, 'messages', 'en');
+
+const MESSAGE_MULTI_DOMAINS_MULTI_LOCALES_WITH_PARAMETERS: Message<{ foobar: { parameters: { parameter1: string } }, messages: { parameters: { parameter2: string } } }, 'fr' | 'en'> = {
+    id: 'message.multi.domains.multi.locales.with.parameters',
+    translations: {
+        foobar: {
+            fr: 'Un simple message {parameter1}',
+            en: 'A simple message {parameter1}',
+        },
+        messages: {
+            fr: 'Un simple message {parameter2}',
+            en: 'A simple message {parameter2}',
+        }
+    }
+};
+
+trans(MESSAGE_MULTI_DOMAINS_MULTI_LOCALES_WITH_PARAMETERS, {parameter1: 'foo', parameter2: 'bar'});
+trans(MESSAGE_MULTI_DOMAINS_MULTI_LOCALES_WITH_PARAMETERS, {parameter1: 'foo', parameter2: 'bar'}, 'messages');
+trans(MESSAGE_MULTI_DOMAINS_MULTI_LOCALES_WITH_PARAMETERS, {parameter1: 'foo', parameter2: 'bar'}, 'messages', 'fr');
+trans(MESSAGE_MULTI_DOMAINS_MULTI_LOCALES_WITH_PARAMETERS, {parameter1: 'foo', parameter2: 'bar'}, 'messages', 'en');
```

Comment:
What's the purpose of this file?

Reply:
This file was here to quickly tests `trans()` types, like we do in https://github.com/DefinitelyTyped/DefinitelyTyped (for example with [types for `assert`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/assert/assert-tests.ts)), but it's useless since https://github.com/JoshuaKGoldberg/eslint-plugin-expect-type is not installed.

Anyway, I've removed this file in favor of `src/Translator/assets/test/translator.test.ts`, this way we can have "real" + types test at the same time